eng/restore/Directory.Build.targets

```diff
@@ -2,6 +2,7 @@
   <Import Project="..\..\Directory.Build.targets" />
 
   <PropertyGroup>
+    <NETCoreAppMaximumVersion>5.0</NETCoreAppMaximumVersion>
```

Comment:
Whyi s this necessary?

Reply:
This is necessary because Microsoft.Net.SDk sets this property to 3.0 and hence the target which checks that the version we are currently building is not larger than NETCoreAppMaximumVersion fails. so we have to override this manually.

Reply:
Is this temporary until we have a new SDK? If yes please add a comment. Same for the one below.

Reply:
Yes this is temporary and we can remove this after the new sdk. I will add the comment.